src/main/java/by/exadel/internship/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package by.exadel.internship.service.impl;
+
+import by.exadel.internship.service.FileService;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import java.util.UUID;
+
+@Service
+public class FileServiceImpl implements FileService {
+
+    private static final String DOWNLOAD_URL =
+            "https://firebasestorage.googleapis.com/v0/b/gs://internship-project-e202a.appspot.com/o/%s?alt=media";
+    private static final String BUCKET_NAME = "internship-project-e202a.appspot.com";
+
+    @Override
+    public ResponseEntity<String> upload(MultipartFile multipartFile) {
+
+        try {
+            String fileName = multipartFile.getOriginalFilename();                        // to get original file name
+            fileName = UUID.randomUUID().toString().concat(this.getExtension(fileName));  // to generated random string values for file name.
+
+            File file = this.convertToFile(multipartFile, fileName);                      // to convert multipartFile to File
+            String tempURL = this.uploadFile(file, fileName);                                   // to get uploaded file link
+            file.delete();                                                                // to delete the copy of uploaded file stored in the project folder
+            return new ResponseEntity<>(tempURL, HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Server Error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+
+    private String uploadFile(File file, String fileName) throws IOException {
+        BlobId blobId = BlobId.of(BUCKET_NAME, fileName);
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("media").build();
+        Credentials credentials = GoogleCredentials.fromStream(new FileInputStream("internship-cloud/internship-project-e202a.json"));
```

Comment:
Should be changable. Get it for example from env var.

Reply:
Done